### PR TITLE
*: enable the new collation by default

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: ref #32015

Problem Summary:
enabled the new collation by default
### What is changed and how it works?
change the default value and fix tests
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
The new collation is enabled by default
```

### CODE REVIEW

**Review by xiongjiwei — APPROVED** [review]

File: config/config.toml.example [hunk 1]
```diff
@@ -109,7 +109,7 @@ repair-table-list = []
 max-server-connections = 0
 
 # Whether new collations are enabled, as indicated by its name, this configuration entry take effect ONLY when a TiDB cluster bootstraps for the first time.
-new_collations_enabled_on_first_bootstrap = false
+new_collations_enabled_on_first_bootstrap = true
```
Comment (ichn-hu) [inline config/config.toml.example:112]: the naming of this configuration is different than any other configurations, is it more appropriate to rename it to

new-collations-enabled-on-first-bootstreap

Comment (wjhuang2016) [reply config/config.toml.example:112]: It's a sad story, we want to fix it, but not in this PR.

File: expression/integration_test.go [hunk 2]
```diff
@@ -2776,6 +2777,9 @@ func TestTiDBDecodeKeyFunc(t *testing.T) {
 	store, clean := testkit.CreateMockStore(t)
 	defer clean()
 
+	collate.SetNewCollationEnabledForTest(false)
+	defer collate.SetNewCollationEnabledForTest(true)
```
Comment (tangenta) [inline expression/integration_test.go:2781]: Why did the test fail if new collation is enabled?

Comment (wjhuang2016) [reply expression/integration_test.go:2781]: EncodeKey depends on the collation, the previous result is based on the old collation

**Review by tangenta — APPROVED** [review]

**Comment by bb7133** [comment]
Please fix the test case: https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/34012/pipeline:

```
FAIL: executor_test.go:5728: testSuiteWithCliBaseCharset.TestCharsetFeatureWithoutNewCollation
```

**Comment by zhangjinpeng1987** [comment]
How about the cluster upgrade from old version?

**Comment by wjhuang2016** [comment]
> How about the cluster upgrade from old version?

It has nothing to do with the old cluster. This change only affects the brand-new cluster.

**Review by morgo — APPROVED** [review]
Config package changes LGTM

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**